solidity/contracts/PhasedEscrow.sol

```diff
@@ -71,3 +71,82 @@ contract CurveRewardsEscrowBeneficiary is Ownable {
         curveRewards.notifyRewardAmount(amount);
     }
 }
+
+/// @dev Interface of recipient contract for approveAndCall pattern.
+interface IStakerRewards { function receiveApproval(address _from, uint256 _value, address _token, bytes calldata _extraData) external; }
+
+/// @title BeaconBackportRewardsEscrowBeneficiary
+/// @notice A beneficiary contract that can receive a withdrawal phase from a
+///         PhasedEscrow contract. Immediately stakes the received tokens on a
+///         designated BeaconBackportRewardsEscrowBeneficiary contract.
+contract BeaconBackportRewardsEscrowBeneficiary is Ownable {
+    IERC20 public token;
+    IStakerRewards public stakerRewards;
+
+    constructor(IERC20 _token, IStakerRewards _stakerRewards) public {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+
+    function __escrowSentTokens(uint256 amount) external {
+        token.approve(address(stakerRewards), amount);
+        stakerRewards.receiveApproval(address(this), amount, address(token), "");
+    }
+}
+
+/// @title BeaconRewardsEscrowBeneficiary
+/// @notice A beneficiary contract that can receive a withdrawal phase from a
+///         PhasedEscrow contract. Immediately stakes the received tokens on a
+///         designated BeaconRewardsEscrowBeneficiary contract.
+contract BeaconRewardsEscrowBeneficiary is Ownable {
+    IERC20 public token;
+    IStakerRewards public stakerRewards;
+
+    constructor(IERC20 _token, IStakerRewards _stakerRewards) public {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+
+    function __escrowSentTokens(uint256 amount) external {
+        token.approve(address(stakerRewards), amount);
+        stakerRewards.receiveApproval(address(this), amount, address(token), "");
```

Comment:
Not that I expect any problems here, but I think it's worth calling `receiveApproval` only if `approve` [returned `true`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/85334995073fff746961155113b0906b689a044c/contracts/token/ERC20/ERC20.sol#L134).

Reply:
Done.